xarray/tests/test_datatree_mapping.py

```diff
@@ -184,100 +69,80 @@ def minmax(ds):
     def test_return_wrong_type(self, simple_datatree):
         dt1 = simple_datatree
 
-        @map_over_datasets
-        def bad_func(ds1):
-            return "string"
-
-        with pytest.raises(TypeError, match="not Dataset or DataArray"):
-            bad_func(dt1)
+        with pytest.raises(
+            TypeError,
+            match=re.escape(
+                "the result of calling func on the node at position is not a "
+                "Dataset or None or a tuple of such types"
+            ),
+        ):
+            map_over_datasets(lambda x: "string", dt1)  # type: ignore[arg-type,return-value]
 
     def test_return_tuple_of_wrong_types(self, simple_datatree):
         dt1 = simple_datatree
 
-        @map_over_datasets
-        def bad_func(ds1):
-            return xr.Dataset(), "string"
-
-        with pytest.raises(TypeError, match="not Dataset or DataArray"):
-            bad_func(dt1)
+        with pytest.raises(
+            TypeError,
+            match=re.escape(
+                "the result of calling func on the node at position is not a "
+                "Dataset or None or a tuple of such types"
+            ),
+        ):
+            map_over_datasets(lambda x: (x, "string"), dt1)  # type: ignore[arg-type,return-value]
 
     @pytest.mark.xfail
     def test_return_inconsistent_number_of_results(self, simple_datatree):
         dt1 = simple_datatree
 
-        @map_over_datasets
-        def bad_func(ds):
-            # Datasets in simple_datatree have different numbers of dims
-            # TODO need to instead return different numbers of Dataset objects for this test to catch the intended error
-            return tuple(ds.dims)
+        # Datasets in simple_datatree have different numbers of dims
+        # TODO need to instead return different numbers of Dataset objects fo
+        # this test to catch the intended error
 
         with pytest.raises(TypeError, match="instead returns"):
-            bad_func(dt1)
+            map_over_datasets(lambda ds: tuple(ds.dims), dt1)
```

Comment:
This change should make this test work now shouldn't it? So it can be un-xfailed?

Reply:
done